plugins/API/DataTable/MergeDataTables.php

```diff
@@ -25,13 +24,14 @@ public function mergeDataTables($table1, $table2)
     {
         // handle table arrays
         if ($table1 instanceof DataTable\Map && $table2 instanceof DataTable\Map) {
-            $subTables2 = $table2->getDataTables();
-            foreach ($table1->getDataTables() as $index => $subTable1) {
-                if (!array_key_exists($index, $subTables2)) {
-                    // occurs when archiving starts on dayN and continues into dayN+1, see https://github.com/piwik/piwik/issues/5168#issuecomment-50959925
-                    continue;
+            $subTables1 = $table1->getDataTables();
+            foreach ($table2->getDataTables() as $index => $subTable2) {
+                if (!array_key_exists($index, $subTables1)) {
+                    $subTable1 = $this->makeNewDataTable($subTable2);
```

Comment:
@diosmosis reading the previous comment in  https://github.com/piwik/piwik/issues/5168#issuecomment-50959925 I wonder if this could cause any issues with potentially wrong data when archiving overlaps another day basically? Not sure I fully understand #5168 .  I suppose if it was a problem, it would need to be handled somewhere else maybe and not in this class as when merging you expect both tables to be merged as it is implemented in this PR

Reply:
So:

- archive starts on one day sending request for 'today'
- new day begins, archive finishes, 'today' value changes

then some bug happens? I'm not sure I understand either.

Reply:
maybe @mattab knows?

Reply:
as you said, "if it was a problem, it would need to be handled somewhere else maybe and not in this class as when merging you expect both tables to be merged as it is implemented in this PR" -> so sounds good to proceed with this proper fix and we can see if the problem in #5168 starts appearing again